validator/specs/lib/validateCmptSpec.js

```diff
@@ -0,0 +1,495 @@
+
+    it('returns false if the cmpt has an invalid pnts inner tile', function() {
+        expect(validateCmpt(createCmptInvalidPnts()).result).toBe(false);
+    });
+
+    it('returns false if the cmpt contains any invalid inner tiles', function() {
+        expect(validateCmpt(createCmptInvalidCombination()).result).toBe(false);
+    });
+});
+
+function createInvalidMagic() {
+    var cmptTile = new Buffer(16);
+    cmptTile.write('xxxx', 0); // magic
+    cmptTile.writeUInt32LE(1, 4); // version
+    cmptTile.writeUInt32LE(cmptTile.length, 8); // byteLength
+    cmptTile.writeUInt32LE(0, 12); // tilesLength
+
+    return cmptTile;
+}
+
+function createInvalidVersion() {
+    var cmptTile = new Buffer(16);
+    cmptTile.write('cmpt', 0); // magic
+    cmptTile.writeUInt32LE(5, 4); // version
+    cmptTile.writeUInt32LE(cmptTile.length, 8); // byteLength
+    cmptTile.writeUInt32LE(0, 12); // tilesLength
+
+    return cmptTile;
+}
+
+function createInvalidByteLength() {
+    var cmptTile = new Buffer(16);
+    cmptTile.write('cmpt', 0); // magic
+    cmptTile.writeUInt32LE(1, 4); // version
+    cmptTile.writeUInt32LE(cmptTile.length - 1, 8); // byteLength
+    cmptTile.writeUInt32LE(0, 12); // tilesLength
+
+    return cmptTile;
+}
+
+function createCmptShort() {
+    var byteOffset = 0;
+    var sizeU32Int = 4;
+    var cmptTile = new Buffer(48);
+
+    //cmpt header
+    cmptTile.write('cmpt', byteOffset); // magic
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(1, byteOffset); // version
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(cmptTile.length, byteOffset); // byteLength
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(5, byteOffset); // tilesLength
+    byteOffset += sizeU32Int;
+
+    //i3dm header
+    cmptTile.write('i3dm', byteOffset); // magic
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(1, byteOffset); // version
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(cmptTile.length - 16, byteOffset); // byteLength
+    byteOffset += sizeU32Int;
+    byteOffset += sizeU32Int; // skip featureTableJSONByteLength
+    byteOffset += sizeU32Int; // skip featureTableBinaryByteLength
+    byteOffset += sizeU32Int; // skip batchTableJSONByteLength
+    byteOffset += sizeU32Int; // skip batchTableBinaryByteLength
+    cmptTile.writeUInt32LE(0, byteOffset); // gltfFormat: 0 - url
+
+    return cmptTile;
+}
+
+function createCmptUnidentifierInner() {
+    var byteOffset = 0;
+    var sizeU32Int = 4;
+    var cmptTile = new Buffer(40);
+
+    //cmpt header
+    cmptTile.write('cmpt', byteOffset); // magic
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(1, byteOffset); // version
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(cmptTile.length, byteOffset); // byteLength
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(1, byteOffset); // tilesLength
+    byteOffset += sizeU32Int;
+
+    //unknown header
+    cmptTile.write('xxxx', byteOffset); // magic
```

Comment:
Provide valid data for the rest of the header so it doesn't trigger a different throw.